### DISCUSSION TITLE
Control with Fill, EnsureViewboxSize, and EnsureViewboxPosition

### DISCUSSION THREAD

**Discussion by johannpistorius** [ElinamLLC/SharpVectors#241] [Q&A]
SharpVectors version : 1.8.1
Framework: .NET Core 3.1
Graphical framework: WPF

# Expected behavior

An svg that has a fill property while also ensuring that the size and position are not altered.

![2022-12-21_11h59_43](https://user-images.githubusercontent.com/15726308/208889916-94bef2e3-3a2b-4f5c-9970-8227e0e64b76.png)

# Actual behavior

![2022-12-21_11h58_05](https://user-images.githubusercontent.com/15726308/208889961-dc49711c-40c1-4a35-adad-d1e84215fdf9.png)

* Left: SvgIcon with Fill property
* Right: SvgViewbox with EnsureViewboxPosition and EnsureViewboxSize set to true

# What I have tried

As per the discussion #189, I used the `SvgViewboxEx` and in my xaml, I had the following

```xml
<local:SvgViewboxEx EnsureViewboxPosition="True" EnsureViewboxSize="True" UriSource="{Binding Path}" Fill="{Binding Fill}"/>
```

Setting the fill while also setting EnsureViewboxPosition and EnsureViewboxSize to true is not working as expected.

# Conclusion

Is there a work around where the Fill, EnsureViewboxPosition, and EnsureViewboxSize properties can be set? Are there any plans to add these properties to either SvgIcon or SvgViewbox?

Thanks

**Comment by paulushub** [comment]
> Are there any plans to add these properties to either SvgIcon or SvgViewbox?

Currently, SvgViewbox supports the  EnsureViewboxPosition, and EnsureViewboxSize properties. But there is no plan to add the Fill property for simplicity. Also, there are plans to extend SvgViewbox to provide interactive effects, so for now it is better to keep it simply until it is completed.

The SvgIcon is the control meant for monochrome SVG and could be extended to support  EnsureViewboxPosition, and EnsureViewboxSize properties - these are meant for icons and the simpler the better and it explains why this class is sealed.

SvgIcon simply extends SvgBitmap, so we could provide a more extensive version of SvgIcon by providing a new extension of SvgBitmap.

**Reply by johannpistorius** [reply]
Thank you for your reply.
For the time being, I'm going to stick with the SvgIcon for the Fill property.
When could we expect a new extensive version of SvgIcon?

**Reply by paulushub** [reply]
> When could we expect a new extensive version of SvgIcon?

The usual issue with new release is testing. Someone just reported an issue when upgrading from 1.8.0 to 1.8.1.
When I finally implemented SvgIcon as part of https://github.com/ElinamLLC/SharpVectors/issues/189, I could hardly get anyone to test and verify it. 
If you are ready to test, then post this as issue and continue to give any feedback on the implementations.
